src/library/scala/collection/Iterator.scala

```diff
@@ -635,14 +635,7 @@ trait Iterator[+A] extends IterableOnce[A] with IterableOnceOps[A, Iterator, Ite
     def next() = if (hasNext) { hdDefined = false; hd } else Iterator.empty.next()
   }
 
-  def drop(n: Int): Iterator[A] = {
-    var i = 0
-    while (i < n && hasNext) {
-      next()
-      i += 1
-    }
-    this
-  }
+  def drop(n: Int): Iterator[A] = sliceIterator(n, -1)
```

Comment:
if this is lazy now, we should add a test for laziness

Reply:
It was lazy before and the unit tests are still there. It wasn't testing anything for a while.

Reply:
@som-snytt sorry I don't understand what you're saying. There is a change here, `LazyList(1 to 10: _*).map(x => {println(x); x}).iterator.drop(4)` used to evaluate the first 4 elements, now it waits for `hasNext`.

But I assume this doesn't break any guarantees?

Reply:
For the record, `LazyList(1 to 10: _*).map(x => {println(x); x}).iterator.slice(4, 10)` also doesn't evaluate anything until `hasNext`. So the change here is either an improvement or neutral.

Reply:
I don't remember what I was thinking before the last busy two weeks. I did not follow up yet on improved lawfulness testing, which is required for membership in the collections collective, so I'll add this to things-to-test. I may have had one of these in mind: 
https://github.com/scala/scala/blob/2.13.x/test/junit/scala/collection/IteratorTest.scala#L33 
https://github.com/scala/scala/blob/2.13.x/test/junit/scala/collection/IteratorTest.scala#L208

Reply:
I guess by "before", I meant "before 2.13", not "before this change". But I haven't reviewed the tests again yet.